src/Orleans.Runtime/Messaging/MessageCenter.cs

```diff
@@ -117,6 +119,19 @@ public void PrepareToStop()
         {
         }
 
+        private void WaitToRerouteAllQueuedMessages()
+        {
+            DateTime maxWaitTime = DateTime.UtcNow + this.messagingOptions.RerouteQueuedMessagesOnSiloShutdownTimeout;
+            while (DateTime.UtcNow < maxWaitTime)
+            {
+                var applicationMessageQueueLength = this.OutboundQueue.ApplicationMessageCount;
+                if (applicationMessageQueueLength == 0)
+                    break;
+                Task.Delay(TimeSpan.FromSeconds(1)).Wait();
```

Comment:
If we are blocking the thread anyway, there's no need to use Task.Delay(). Can do Thread.Sleep() instead.

Also, we might want to reduce the iteration wait time here to lower than a second because, hopefully, in most cases forwarding will be very fast, especially in test scenarios. Maybe 100 milliseconds instead?